### ISSUE TITLE
Stuck Loading Smart Connections on Linux Obsidian

### ISSUE THREAD

**Issue by davidotron** [brianpetro/obsidian-smart-connections#750]
I remember this working on my Windows machine. After switching to the Obsidian flatpak on Linux Mint, I've re-installed Smart Connections, and the module is stuck "Loading Smart Connections..." forever (on this same status for an hour now)

![image](https://github.com/user-attachments/assets/4c6bc87a-46fd-4613-b95a-150885f4a825)

Also, maybe related, the model platform for Smart Chat is not available:

![image](https://github.com/user-attachments/assets/2fd337c2-e01c-49a7-aa2d-0b1f17e336bd)

If I click that drop-down square, nothing loads. I have a local OpenAPI endpoint on my network I'd like to use here.

Is this a known issue in the Linux version of Obsidian, and/or is there a solution here?

Thank you!!

**Comment by Nesador95** [comment]
I can say for certain that this exact behavior is also happening in my windows install, it's not your OS, its the latest patch that introduced the bug. everything was working fine and when I updated the plug-in with today's release 2.1.96, that's when things broke. 
![image](https://github.com/user-attachments/assets/cea4a102-a6d9-421b-bfaf-b36e2e073a68)
It's stuck there and wont make embeddings, additionally it won't allow me to set up my chatbot as you described above.

**Comment by brianpetro** [comment]
Thanks for reporting this issue.

Please open the developer console logs and screenshot any errors (should be highlighted in red) that you see there and I will check it out 🌴

**Comment by davidotron** [comment]
@brianpetro thanks for responding so quickly!!

Developer console logs below upon **enabling the plugin**. Let me know if you'd like to see anything else.

![image](https://github.com/user-attachments/assets/cafccd04-6de4-4b0d-9c14-aa3bb0d6f098)

**Comment by brianpetro** [comment]
It seems like using the GPU might be the issue. To disable using it, in the settings there should be a `gpu_batch_size` setting. Set this to `0` and it should prevent the GPU from being used. Then restart the Smart Connections and see if there's any improvement 🌴

**Comment by davidotron** [comment]
No luck unfortunately. :( I set GPU batch size to 0, reloaded the plugin and reloaded the Obsidian app, same result. Fresh console logs below:

![image](https://github.com/user-attachments/assets/15c176dd-5761-45da-b20d-b4e840bb8635)

I was thinking that hopefully the GPU wasn't the issue, I have an RTX 3080 with a pretty stable driver version and it'd be nice to use local GPU for some things.

**Comment by brianpetro** [comment]
@davidotron thanks for sharing that. 

I'm pretty sure this is an issue that we ran into with the Smart Connect app. Unfortunately, the only way to fix it would be to update configurations in the Obsidian source code.

However, I was able to make those changes in the Smart Connect app. I'm still working on making the processing embeddings UI in Smart Connect, but soon you will be able to use Smart Connect app to manage the embedding in the background. Then all the Obsidian plugin will need to do is load those embeddings.

Did the embedding work for before the latest version?

@Nesador95 you may be experiencing a different issue. Please screenshot the console logs and let me know. 

🌴

**Comment by davidotron** [comment]
@brianpetro yes, but I can only speak to it working on Windows a while back. I recently went through a big conversion in my life from Windows to Linux Mint and stopped using Obsidian for a few weeks. Long story.

I can say however this used to work, at least on the Windows version. Thanks for working on it.

**Comment by davidotron** [comment]
@brianpetro Ok, I was able to roll back to 2.1.95 and it looks to be working now, similar to the other Linux guy.

**Comment by MartinHuber2** [comment]
Same problem with v2.1.97. How can I go back to v2.1.95 ?

**Comment by davidotron** [comment]
@MartinHuber2 Download the .95 release here: https://github.com/brianpetro/obsidian-smart-connections/releases/tag/2.1.95

In your vault folder, navigate to .obsidian/plugins. Make a backup of the smart-connections plugin folder and delete it. make a new smart-connections folder and extract all the files (main, manifest, and styles) there. Reboot Obsidian and go to your plugins; you should see Smart Connections on 2.1.95 now.

You may be able to copy the Data settings to the new Data file depending on what's changed between versions, but I just re-configured everything in the Obsidian plugin UI.

**Comment by MartinHuber2** [comment]
Thx for the rapid response. I am on 2.1.95 now, bit still stuck with "Loading smart connections...", and no options under "Model Platform". 
![image](https://github.com/user-attachments/assets/9a60bb6a-07f8-4735-9643-33df22077e6d)

**Comment by christianngarcia** [comment]
Thanks for the screenshot. If you can also screenshot any errors you see in the developer console logs that would be a big help in figuring out the issue. 😊

**Comment by brianpetro** [comment]
@MartinHuber2, the model platform settings should appear if you re-click the Smart Connections settings. Unfortunately, the load time is delayed currently, which is why it isn't fully rendering every time. This will be addressed in the upcoming improvements to the Smart Chat 🌴

**Comment by brianpetro** [comment]
If local embeddings broke in/after `v2.1.96`, see https://github.com/brianpetro/obsidian-smart-connections/issues/749#issuecomment-2324472086 🌴

**Comment by jason-ruis** [comment]
I'm having a similar issue at the moment. 
![Screenshot 2024-09-04 at 11 22 31 AM](https://github.com/user-attachments/assets/c815a40a-1f5f-4436-8b60-5ff87ac05ef7)

**Comment by Korayem** [comment]
same on v2.1.100

**Comment by thgie** [comment]
- Debian 12
- Obsidian 1.6.7
- Smart Connections 2.2.76

I had the same problem with being stuck at "Loading Smart Connections" after the initial scan. The only thing I had to do was to toggle the "Legacy Transformers (no GPU)" off and on again. Then it started to calculate the embeddings.

**Comment by JoaoCarlosP** [comment]
> * Debian 12
> * Obsidian 1.6.7
> * Smart Connections 2.2.76
> 
> I had the same problem with being stuck at "Loading Smart Connections" after the initial scan. The only thing I had to do was to toggle the "Legacy Transformers (no GPU)" off and on again. Then it started to calculate the embeddings.

Yes! This works for me too... but after sending a message, it displays a problem. See more in the image:
![image](https://github.com/user-attachments/assets/49b5a79f-d0c9-4df5-9488-08a4c638b610)

@brianpetro

**Comment by brianpetro** [comment]
@JoaoCarlosP  The error says that the API key is past its rate limit (its using a personal key I include that allows 100/requests per day across all users, intended for new users to try out the chat), so you have to add an API key to the settings in the top right corner of the chat window 🌴

**Comment by bydb** [comment]
Currently, I am trying to use the smart chat module. I have tested both OpenAI with an API key and the local Ollama, but the module keeps saying that it is using the default model. How can I make smart chat use OpenAI?
<img width="372" alt="Bildschirmfoto 2024-11-01 um 11 25 15" src="https://github.com/user-attachments/assets/ed1bad26-2719-48a0-aaa1-3c477bf0f736">
<img width="323" alt="Bildschirmfoto 2024-11-01 um 11 00 34" src="https://github.com/user-attachments/assets/e8bf0916-f7eb-4adc-b557-667677515697">

**Comment by brianpetro** [comment]
Thanks for letting me know.

It might be a bug. If it is, uninstalling the plugin and reinstalling might fix it as long as you reconfigure the chat settings before the notification appears again.

🌴

**Comment by bydb** [comment]
I have installed and uninstalled the module several times, but unfortunately, nothing has changed.

**Comment by brianpetro** [comment]
Good to know.

I'll have to look into it further 🌴

**Comment by osaze-nosaze** [comment]
Hi, I am having the "smart environment is loading..." problem. Uninstalling and reinstalling several times has not helped. Neither has returning to earlier versions of Smart Connections. Neither also has disabling all other plugins. Smart Connections worked fine until recently. I noticed that it was not indexing my vault, so I went to "Settings" to see what the trouble was. The problem popped up as soon as I clicked the options button for Smart Connections. This has happened every time. The panel says "smart environment is loading...", and stays like that indefinitely. So I cannot get to the options to change the plugin settings or do anything else about it. 

I'd appreciate help with this. Below is the text from the error console:

index.html:1 Uncaught (in promise) SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at SmartEnv2.load_settings (plugin:smart-connections:1307:31)
    at async SmartSettings.load (plugin:smart-connections:426:27)
    at async SmartSettings.create (plugin:smart-connections:356:5)
    at async SmartEnv2.load (plugin:smart-connections:1031:25)
    at async SmartEnv2.load (plugin:smart-connections:22342:5)
    at async eval (plugin:smart-connections:1000:7)

Thank you in advance.

**Comment by brianpetro** [comment]
Try deleting the `.smart-env/settings.json` file

Then restarting Obsidian 

🌴

**Comment by osaze-nosaze** [comment]
hi, @brianpetro. Thanks for the prompt response. Happily, I stumbled on the solution to the problem---purely by chance, by trial-and-error, by "why not; what have I to lose?" I found that the .smart-env folder was last modified on 22 December 2025 whereas the date for the smart-connections folder was 16 February 2026. That might be the source of the problem, I thought. So I experimentally deleted the former, hoping that smart-connections would recreate it when I restarted Obsidian. Luckily, that was what happened and it fixed the problem. So your suggestion---a less drastic measure---probably would have achieved the same end. Others with this problem might benefit from this.
Thank you very much for a great plugin and problem resolution.
